test/interfaces.jl

```diff
@@ -0,0 +1,113 @@
+#using QEDprocesses
+using Random
+using Distributions
+using QEDbase
+
+RNG = MersenneTwister(708583836976)
+DIM = rand(RNG, 2:8)
+
+ATOL = 0.0
+RTOL = sqrt(eps())
+
+RAND_MAX = rand(RNG)
+RAND_EXACTNESS = rand(RNG, Bool)
+
+struct TestSetup{D} <: QEDevents.AbstractComputationSetup
+    dist::D
+end
+
+function TestSetup(dim::Integer)
+    return TestSetup(product_distribution(Uniform.(-rand(dim), 1.0)))
+end
+
+Base.size(stp::TestSetup, N::Integer) = size(stp.dist)[1] ###best way?
+Base.size(stp::TestSetup) = size(stp, 1) #??
+
+QEDevents._compute(stp::TestSetup, x) = pdf(stp.dist, x)
+
+struct TestSampler <: AbstractSampler
+    stp::TestSetup
+end
+
+Base.size(samp::TestSampler, N::Integer) = size(samp.stp, N)
+Base.size(samp::TestSampler) = size(samp, 1)
+
+function QEDevents._rand!(
+    rng::AbstractRNG, smplr::TestSampler, res::AbstractVector{P}
+) where {P}
+    rand!(rng, smplr.stp.dist, res)
+    return res
+end
+
+is_exact(::TestSampler) = RAND_EXACTNESS
+
+@testset "sampler interface" begin
+    setup(smplr::TestSampler) = smplr.stp
+    QEDevents._weight(smplr::TestSampler, x) = QEDevents.compute(smplr.stp, x) # assuming a product of Uniforms    ##remove the QEDevents nstead of replacing by processes because exported
+    max_weight(smplr::TestSampler) = pdf(smplr.stp.dist, zeros(size(smplr.stp.dist)))
+    function Distributions._rand!(
+        rng::Random.AbstractRNG, s::TestSampler, x::AbstractVector{T}
+    ) where {T<:Real}
+        # assuming a sampler setup from Distributions.jl
+        return Distributions.rand!(rng, s.stp.dist, x)
+    end
+
+    @testset "process sampler interface" begin
+        proc_stp = TestSetup(DIM)
+        test_smplr = TestSampler(proc_stp)
+
+        @testset "properties" begin
+            @test setup(test_smplr) == proc_stp
+            @test is_exact(test_smplr) == RAND_EXACTNESS
+            @test size(test_smplr) == size(proc_stp)
```

Comment:
`DIM?`

Reply:
Implemented in 40475f80c9b50a7d4831a5316723372b2a6c56a9